packages/amplication-server/src/core/subscription/subscription.controller.ts

```diff
@@ -0,0 +1,63 @@
+import { Body, Controller, Post } from "@nestjs/common";
+import { BillingService } from "../billing/billing.service";
+import { CreateSubscriptionInput } from "./dto/CreateSubscriptionInput";
+import { SubscriptionData } from "./dto/SubscriptionData";
+import { UpdateStatusDto } from "./dto/UpdateStatusDto";
+import { UpdateSubscriptionInput } from "./dto/UpdateSubscriptionInput";
+import { SubscriptionService } from "./subscription.service";
+
+@Controller("subscriptions")
+export class SubscriptionController {
+  constructor(
+    private readonly subscriptionService: SubscriptionService,
+    private readonly billingService: BillingService
+  ) {}
+
+  @Post("updateStatus")
+  async updateStatus(@Body() updateStatusDto: UpdateStatusDto): Promise<void> {
+    switch (updateStatusDto.type) {
```

Comment:
are we using any type of auth on this endpoint?

Reply:
Sure, I forgot to add it. Please, see the updated version.